Added new icon: "\ue90e" is MICROSD

jdlcdl opened the pull request:
DRAFT

Update Wed Aug 2nd:  This draft will be closed soon,
because of easyuxd's responsiveness, and because we all know that the following text is truth. 

BEWARE THE PROGRAMMER WHO DABBLES IN GRAPHIC-DESIGN!!!

related to #344 and pr #410:
I wanted a new icon, because the SDCARD icon hangs low, creating kerning conflicts on the following line.  Added a MICROSD icon to the seedsigner custom icon set, but I have no clue what I might have broken in this fontset.
This would need review by an expert, I'm just leaving this here so that others could take a look.

![fontviewer_screenshot](https://github.com/SeedSigner/seedsigner/assets/117163651/016f3e0b-0079-433d-b0a9-2c63feaf1cc0)

I've altered the filename to use uppercase, just in case it gets merged when it truly should not!!!

FYI: I never succeeded in editing the otf file currently in use, to add just one more icon.  Instead, I used fontforge to create a new font set, imported the existing 12 glyphs (previously exported as eps), added one more, and generated as OTF.  I may have broken lots.

Thanks to Keith's screenshot_generator, we can see what some of our screens would look like using the new SeedSigner-Glyphs.otf fonts file [here](https://github.com/jdlcdl/seedsigner-screenshots/blob/screenshots_new_SeedSigner-Glyphs/screenshots_new_SeedSigner-Glyphs/README.md).

jdlcdl commented:
Just a note that:

This pr got some positive feedback from Nick at today's meeting.

While I was able to make all of the screens in the link above labeled "here", I still think it's worth a designer to take a look the glyphs file (or at least whoever it was that used icomoon to create the existing seedsigner-glyphs.otf file).

THIS PR WOULD REQUIRE ONE MORE COMMIT, so that the filename over-writes the existing fonts file.

newtonick commented:
@easyuxd are you able to look at the new otf file on this PR. I know very little about otf font files. When I validate it with FontBook on my Mac I do get one error. "'name' table structure" is all it says. Not sure what that means. I also noticed the otf contains only 14 glyph count instead of 16 like the original: https://github.com/SeedSigner/seedsigner/blob/dev/src/seedsigner/resources/fonts/seedsigner-glyphs.otf

I'm basically looking for feedback in 2 ways.
1) is there anything fishy or wrong about font otf?
2) are there any improvements or changes you would recommend based on why you know?

Thanks!

jdlcdl commented:
I also noticed the otf contains only 14 glyph count instead of 16 like the original: https://github.com/SeedSigner/seedsigner/blob/dev/src/seedsigner/resources/fonts/seedsigner-glyphs.otf

I confirm that when I recreated the new fonts file in this pr, I did NOT include the empty glyphs at U+E905 and U+E906 and that I re-used U+E90E for the new MICROSD glyph which had been empty too.  Another 'oddness' that I did, similar to the existing fonts file, I had to stretch both up/down chevrons to 2048 units whereas the rest were at 1024units.

easyuxd commented:
@jdlcdl I love this, thank you.

We have an [existing SD card icon from FontAwesome](https://fontawesome.com/icons/sd-card?f=classic&s=solid), where we've been sourcing most of the current SeedSigner icon set. If it's okay with you, I'd prefer to keep consistency with the current icon style.

That said, I created a new MicroSD icon for comparison. While I prefer the SD card for it's simpler and recognizable shape (esp on a small screen), if we feel the MicroSD icon is more precise I can export it instead. Thoughts?

![sd-vs-microsd](https://github.com/SeedSigner/seedsigner/assets/86452817/35e2bfd4-acc2-4e4c-a253-c66cc9c306de)

---

I've had the SD card icon in my design file for a long time, but I don't think it made it into the OTF file. In fact, I have more custom icons that I was meaning to complete as a set. If we can nail down a process for packaging up the OTF icon fonts, I'd be happy to maintain the file (or just maintain the SVGs and add them to the [design system](https://github.com/easyuxd/seedsigner-design-system#iconography) and others can convert them).

For context, @kdmukai and I had worked on this back in March 2022, and we had issues getting the icons to render properly using FontForge. The current OTF was created with the [IcoMoon](https://icomoon.io/) converter, with icons starting at U+E900.

jdlcdl commented:
I started this pr with advice, in all caps, that I've known and been a believer in for a long time.  I would always prefer an expert to deal with fonts.

My only issue was the height of the MICROSD hung "below the line" like the tails of lowercase gjpqy and it was interfering with a colored line of text just below.  as in this image:
https://user-images.githubusercontent.com/117163651/256928890-e160c651-142a-4028-b864-77842dbafef3.png

Beyond that, I would recommend whatever you recommend.

newtonick commented:
> That said, I created a new MicroSD icon for comparison. While I prefer the SD card for it's simpler and recognizable shape (esp on a small screen), if we feel the MicroSD icon is more precise I can export it instead. Thoughts?

I do like the icon on the left better.


> My only issue was the height of the SDCARD hung "below the line" like the tails of lowercase gjpqy and it was interfering with a colored line of text just below. as in this image:
> https://user-images.githubusercontent.com/117163651/256928890-e160c651-142a-4028-b864-77842dbafef3.png

I wonder if the best solution is to update WarningScreen to allow passing in the status_icon_size parameter from LargeIconStatusScreen as to not overlap the line of text below.

jdlcdl commented:
> I wonder if the best solution is to update WarningScreen to allow passing in the status_icon_size parameter from LargeIconStatusScreen as to not overlap the line of text below.

We can already pass in a status_icon_size since LargeIconStatusScreen will pick it up, and it works to resize the icon, but it doesn't help to make the lack of vertical space between the icon and the next line appear to be any bigger.  As a rule of thumb, I lean away from changing any micro-apis, and I hesitate even more to change the visual layout of those base screens like altering line-height.

newtonick commented:
> We can already pass in a status_icon_size since LargeIconStatusScreen will pick it up, and it works to resize the icon, but it doesn't help to make the lack of vertical space between the icon and the next line appear to be any bigger. As a rule of thumb, I lean away from changing any micro-apis, and I hesitate even more to change the visual layout of those base screens like altering line-height.

That makes sense to me too and agree with your statements.

easyuxd commented:
@jdlcdl 
I think I missed some context that this was to be used in a dialog screen, apologies for that. I appreciate the amount of detail and problem solving you’ve done here. 

Can you test this microsd.otf? I'm not seeing where the icon dips below the baseline.

[microsd.otf.zip](https://github.com/SeedSigner/seedsigner/files/12224393/microsd.otf.zip)

jdlcdl commented:
@easyuxd This icon works perfectly!

![remove_inserted_microsd12](https://github.com/SeedSigner/seedsigner/assets/117163651/217fdeea-9ffb-44a4-a88a-8a9a8d6b8a8b)

When you have time, would you please consider the following:
* That we would need to replace the existing seedsigner-glyphs.otf fonts file [here](https://github.com/SeedSigner/seedsigner/blob/dev/src/seedsigner/resources/fonts/seedsigner-glyphs.otf),
* I had to move your icon to an unused unicode point, importing via fontforge, but I'd much prefer that the entire fonts file is prepared how you feel would be best.
* I noticed that your new icon perfectly aligns with the top and bottom of all others in that file.  I suspect that many icons in the FontAwesome set all hang low and are not the best choice for dialogs that use text immediately beneath (as do some of our screens).  If you still prefer the SDCARD icon without the hard-to-see microsd cut-outs, then I do agree. My preference for this new icon is its real-estate placement rather than the shape.
* We are still undecided about the exact Title, Sub-Icon Text + color, and Body Text for this screen (which will be seen once per session for those who cannot remove the microsd, will be part of the initial-impression for noobs, and will be mostly avoided by opsec savvy veterans).  Any input you might have would be greatly appreciated.  I've been keeping samples in issue #344

If you were to provide a replacement for the existing seedsigner-glyphs.otf file, I would be happy to generate screen shots of our existing screens, as I did [here](https://github.com/jdlcdl/seedsigner-screenshots/blob/screenshots_new_SeedSigner-Glyphs/screenshots_new_SeedSigner-Glyphs/README.md) so that we could all take a look before I ready this pr from "draft" to "needs review".

easyuxd commented:
> That we would need to replace the existing seedsigner-glyphs.otf fonts file [here](https://github.com/SeedSigner/seedsigner/blob/dev/src/seedsigner/resources/fonts/seedsigner-glyphs.otf),

Ok, I'll work on packaging up a new OTF file. I'll probably create a new PR for the update, since I'll be added a number of new icons that will require implementation.

> I noticed that your new icon perfectly aligns with the top and bottom of all others in that file. I suspect that many icons in the FontAwesome set all hang low and are not the best choice for dialogs that use text immediately beneath (as do some of our screens). If you still prefer the SDCARD icon without the hard-to-see microsd cut-outs, then I do agree. My preference for this new icon is its real-estate placement rather than the shape.

Thanks for the input! I'll keep it like the others in seedsigner-glyphs.otf that have good alignment, and as we gradually transition away from FontAwesome hopefully becomes less and less an issue.

> We are still undecided about the exact Title, Sub-Icon Text + color, and Body Text for this screen (which will be seen once per session for those who cannot remove the microsd, will be part of the first-impression for noobs, and will be mostly avoided by opsec savvy veterans). Any input you might have would be greatly appreciated. I've been keeping samples in issue [Feature Request: reminder to pull SD after boot #344](https://github.com/SeedSigner/seedsigner/issues/344)

I was just thinking about this. I added some feedback in that PR.

jdlcdl commented:
> Ok, I'll work on packaging up a new OTF file. I'll probably create a new PR for the update, since I'll be added a number of new icons that will require implementation.
> ...
> I was just thinking about this. I added some feedback in that PR.

Given your input, I have removed the "MicroSD is Inserted!" text just beneath the icon, which means that there is no longer any lack-of-vertical-space between the SDCARD and text.

If that solution gets accepted, we would be much less pressed for your latest icon set;  but I suspect that it will still be greatly appreciated when you have time to do so.  I'm looking forward to seeing it!

jdlcdl commented:
closed because @easyuxd already mentioned a future pr to update fonts.
